packager/amp_cache_transform/amp_cache_transform.go

```diff
@@ -71,13 +80,102 @@ func parseParameterisedIdentifier(reader *strings.Reader) (*parameterisedIdentif
 	if err != nil {
 		return nil, errors.Wrap(err, "parsing primary identifier")
 	}
-	// NOTE: The initial version of AMP-Cache-Transform does not support
-	// any parameters, so we don't bother to implement a parser for them.
-	// No point in distinguishing syntactic failure from semantic failure.
-	// Once we add parameters, we'll need to implement parsers for the
-	// subset item that they use. See Items here:
-	// https://tools.ietf.org/html/draft-ietf-httpbis-header-structure-07#section-3.4
-	return &parameterisedIdentifier{primaryID}, nil
+	// NOTE: The current version of AMP-Cache-Transform only uses
+	// string-valued parameters. Future versions of the spec may require
+	// parsing other Item types.
+	params := map[string]string{}
+	for {
+		if err := discardOWS(reader); err != nil {
+			return nil, errors.Wrap(err, "discarding OWS")
+		}
+
+		// If we've reach the end of the parameter list, exit success.
+		if reader.Len() == 0 {
+			break
+		}
+		semicolon, err := reader.ReadByte()
+		if err != nil {
+			return nil, errors.Wrap(err, "reading ';'")
+		}
+		if semicolon != ';' {
+			if err := reader.UnreadByte(); err != nil {
+				return nil, errors.Wrap(err, "unreading ';'")
+			}
+			break
+		}
+
+		// Otherwise, parse a parameter name.
+		if err := discardOWS(reader); err != nil {
+			return nil, errors.Wrap(err, "discarding OWS")
+		}
+		name, err := parseIdentifier(reader)
+		if err != nil {
+			return nil, errors.Wrap(err, "parsing param name")
+		}
+		if _, has := params[name]; has {
+			return nil, errors.Errorf("param %q already present", name)
+		}
+		if name != "v" {
+			return nil, errors.Errorf("invalid AMP-Cache-Transform param %q", name)
+		}
+
+		// ... and a parameter value.
+		// NOTE: The current version of the AMP-Cache-Transform spec
+		// does not use any null-valued parameters. So, for now, a
+		// missing '=' is a parse failure.
+		equals, err := reader.ReadByte()
+		if err != nil {
+			return nil, errors.Wrap(err, "reading '='")
+		}
+		if equals != '=' {
+			return nil, errors.New("expected '='")
+		}
+		// NOTE: In the future, this may need to be parseItem() to
+		// support other Item types:
+		// https://tools.ietf.org/html/draft-ietf-httpbis-header-structure-07#section-4.2.5
+		value, err := parseString(reader)
+		if err != nil {
+			return nil, errors.Wrapf(err, "parsing value for param %s", name)
```

Comment:
Isn't name always 'v'? No need for the string format.

Reply:
OK, I changed this. name can be anything now. Reason is that an invalid param should only fail one identifier, not the whole thing. (Added test.)